docs/best_practices.md

```diff
@@ -0,0 +1,153 @@
+
+The main settings you'll want to replace with variables are in the `[[outputs.influxdb_v2]]` section of your configuration:
+
+```
+urls = ["$INFLUX_HOST"]
+
+token = "$INFLUX_TOKEN"
+
+organization = "$INFLUX_ORG"
+```
+
+If your template allows the user to choose their own `bucket` values, you should make this a variable in your configuration also:
+
+```
+bucket = "$INFLUX_BUCKET"
+```
+
+Depending on what input plugins you use, there will likely be others that you want to make variables, such as host names, databases, or authentication credentials.
+
+> **Note:** Be sure to document all of these variables in the `README.md` for your Template so that users will know what they need to set and what values they should use.
+
+### Telegraf Configuration Labels
+
+Sometimes, it's handy to group Telegraf connfigurations by specific plugins. For example, if I want to make a change to an input plugin, being able to quickly find all the Telegraf configurations using that plugin is helpful. To this end, it's useful to add Labels such as `inputs.plugin_name` to your Telegraf configurations to help users filter in the UI. 
+
+### Adding Telegraf to your Template
+
+To add a Telegraf config file to your template, you will need to edit the template file manually using the following steps:
+
+ 1. Export your template using the command in the [submitting a template](submit_a_template.md) doc.
+   
+ 2. Add the following to the end of the file, being sure to give a name for your Telegraf configuration:
+    ```
+        ---
+        apiVersion: influxdata.com/v2alpha1
+        kind: Telegraf
+        metadata:
+            name: The Name Of Your Configuration
+            config: |
+        ```
+        
+ 3. Copy/paste the content of your configuration file into the lines below what you just added, indenting it all until it is 4 spaces further indented than the `config:` line.
+    
+
+## Testing Your Template
+
+After you export your finished Template, you should always test it to make sure that it applies cleanly, has everything you wanted to include, and that the instructions your provide in your `README.md` have all the steps needed to use it.
+
+The easiest way to get a clean testing environment of InfluxDB is to run a new Docker container (other methods can be found in our [getting started guide](https://v2.docs.influxdata.com/v2.0/get-started/#start-with-influxdb-oss)). You can start a new container with the latest InfluxDB release by running:
+
+```
+ docker run -p 8086:8086 -p 9999:9999 quay.io/influxdb/influxdb:2.0.0-beta
+
+```
+
+After you create a new `Organization` and `Token` in your docker instance (available at http://localhost:9999), you can follow the [instructions for using a Template](./use_a_template.md) to apply yours.
```

Comment:
After you create a new `Organization` and `Token` in your ~docker~ **Docker** instance (available at http://localhost:9999), ~you can~ follow the [instructions for using a Template](./use_a_template.md) to apply your **template**.